From 27e6c84c1f14a8195a3f9fb489c240ecc7a6257d Mon Sep 17 00:00:00 2001
From: Eric Botcazou <ebotcazou@adacore.com>
Date: Mon, 10 Jan 2022 12:40:10 +0100
Subject: [PATCH] Properly enable -freorder-blocks-and-partition on 64-bit
 Windows

The PR uncovered that -freorder-blocks-and-partition was working by accident
on 64-bit Windows, i.e. the middle-end was supposed to disable it with SEH.
After the change installed on mainline, the middle-end properly disables it,
which is too bad since a significant amount of work went into it for SEH.

gcc/
	PR target/103465
	* coretypes.h (unwind_info_type): Swap UI_SEH and UI_TARGET.
---
 gcc/coretypes.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/gcc/coretypes.h b/gcc/coretypes.h
index 406572e947d..732c76f5cbf 100644
--- a/gcc/coretypes.h
+++ b/gcc/coretypes.h
@@ -227,15 +227,17 @@ enum stack_protector {
   SPCT_FLAG_EXPLICIT = 4
 };
 
-/* Types of unwind/exception handling info that can be generated.  */
+/* Types of unwind/exception handling info that can be generated.
+   Note that a UI_TARGET (or larger) setting is considered to be
+   incompatible with -freorder-blocks-and-partition.  */
 
 enum unwind_info_type
 {
   UI_NONE,
   UI_SJLJ,
   UI_DWARF2,
-  UI_TARGET,
-  UI_SEH
+  UI_SEH,
+  UI_TARGET
 };
 
 /* Callgraph node profile representation.  */
-- 
2.34.1

